### DISCUSSION TITLE
admin or admin/users route force sign-out

### DISCUSSION THREAD

**Discussion by stseaman** [ejazahm3d/fullstack-turborepo-starter#10] [Q&A]
Hello,

Thank you for sharing your code with the community.

I am currently building new website using your repo as a starter and I am facing a challenge that I hope you can maybe help with. I made some changes to the database.

The flow:
1) Launch nextjs dev
2) Open browser and hit http://localhost:3000
3) I can see the page, I click on Sign-In (I used Google provider for next-auth)
4) I go through my Google sign-in flow and the page refreshes displaying my Google user name
5) The problem is when I click on the admin home or users route, I see the page display correctly but right after that I get access denied error.

Looking at developer console in Chrome, I can see a call is being made to signout. I do not understand why there is an automatic signout being called. I noticed an error against Sidebar.tsx "ReferenceError: window is not defined".

Hopefully I misconfigured something.

Thank you.

**Comment by stseaman** [comment]
I found something in my app/(admin)/admin/page.tsx, if I add console.log('session', session) after getServerSession() call, the session prints as null. It appears that when I click on a route, the session is lost somehow. If I stay on / the session persists fine.

**Comment by ejazahm3d** [comment]
@stseaman Hi. Hope you are doing great. This looks more related to next-auth. Not sure I can help much here without looking at the code. You can try searching if you or some of your dependencies is using window object. Use the stack trace to track where it is coming from. That will help you debug where the issue exist. :)
